src/main/java/gregtech/api/metatileentity/multiblock/ui/MultiblockUIBuilder.java

```diff
@@ -0,0 +1,982 @@
+     */
+    public MultiblockUIBuilder setWorkingStatusKeys(String idlingKey, String pausedKey, String runningKey) {
+        if (idlingKey != null) this.idlingKey = IKey.lang(idlingKey).style(TextFormatting.GRAY);
+        if (pausedKey != null) this.pausedKey = IKey.lang(pausedKey).style(TextFormatting.GOLD);
+        if (runningKey != null) this.runningKey = IKey.lang(runningKey).style(TextFormatting.GREEN);
+        return this;
+    }
+
+    /**
+     * Adds the max EU/t that this multiblock can use.
+     * <br>
+     * Added if the structure is formed and if the passed energy container has greater than zero capacity.
+     */
+    public MultiblockUIBuilder addEnergyUsageLine(IEnergyContainer energyContainer) {
+        if (!isStructureFormed || energyContainer == null) return this;
+        boolean hasEnergy = getSyncer().syncBoolean(energyContainer.getEnergyCapacity() > 0);
+        if (!hasEnergy) return this;
+
+        long maxVoltage = Math.max(energyContainer.getInputVoltage(), energyContainer.getOutputVoltage());
+        maxVoltage = getSyncer().syncLong(maxVoltage);
+
+        IKey bodyText = KeyUtil.lang(TextFormatting.GRAY,
+                "gregtech.multiblock.max_energy_per_tick",
+                KeyUtil.number(maxVoltage),
+                KeyUtil.voltage(GTValues.VOCNF, maxVoltage));
+
+        var hoverText = KeyUtil.lang(TextFormatting.GRAY,
+                "gregtech.multiblock.max_energy_per_tick_hover");
+
+        addKey(bodyText, hoverText);
+        return this;
+    }
+
+    /**
+     * Adds the max Recipe Tier that this multiblock can use for recipe lookup.
+     * <br>
+     * Added if the structure is formed and if the passed tier is a valid energy tier index for {@link GTValues#VNF}.
+     */
+    public MultiblockUIBuilder addEnergyTierLine(int tier) {
+        if (!isStructureFormed) return this;
+        tier = getSyncer().syncInt(tier);
+        if (tier < GTValues.ULV || tier > GTValues.MAX) return this;
+
+        var bodyText = KeyUtil.lang(TextFormatting.GRAY,
+                "gregtech.multiblock.max_recipe_tier", GTValues.VNF[tier]);
+        var hoverText = KeyUtil.lang(TextFormatting.GRAY,
+                "gregtech.multiblock.max_recipe_tier_hover");
+        addKey(bodyText, hoverText);
+        return this;
+    }
+
+    /**
+     * Adds the exact EU/t that this multiblock needs to run.
+     * <br>
+     * Added if the structure is formed and if the passed value is greater than zero.
+     */
+    public MultiblockUIBuilder addEnergyUsageExactLine(long energyUsage) {
+        if (!isStructureFormed) return this;
+        energyUsage = getSyncer().syncLong(energyUsage);
+        if (energyUsage > 0) {
+            String energyFormatted = TextFormattingUtil.formatNumbers(energyUsage);
+            // wrap in text component to keep it from being formatted
+            var voltageName = KeyUtil.overclock(GTValues.VOCNF, energyUsage);
```

Comment:
Is there a need for this `KeyUtil` method if it is only called here? Can't you just inline the method here?